src/drivers/imx/ipc.c

```diff
@@ -202,8 +202,9 @@ int platform_ipc_init(struct ipc *ipc)
 	/* allocate page table buffer */
 	iipc->dh_buffer.page_table = rzalloc(SOF_MEM_FLAG_USER,
 					     PLATFORM_PAGE_TABLE_SIZE);
-	if (iipc->dh_buffer.page_table)
-		bzero(iipc->dh_buffer.page_table, PLATFORM_PAGE_TABLE_SIZE);
+	if (!iipc->dh_buffer.page_table)
+		sof_panic(SOF_IPC_PANIC_IPC);
+	bzero(iipc->dh_buffer.page_table, PLATFORM_PAGE_TABLE_SIZE);
```

Comment:
`bzero(iipc->dh_buffer.page_table, PLATFORM_PAGE_TABLE_SIZE);` was only called under `if (iipc->dh_buffer.page_table)`. Now it is always called. Is it Intentional?

Reply:
yes - we panic if page table allocation fails and zero it if allocation succeeds.